### PR TITLE
Sb 061 ganerate date

### PR DESCRIPTION
**Pull request by SimonPe25**
Entity Flight have method uncorrect becose he is small without some string

### CODE REVIEW

File: src/main/java/org/booking/Main.java [hunk 1]
```diff
@@ -1,5 +1,12 @@
 package org.booking;
 
+import org.booking.entity.Flight;
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/Main.java]: 3  +import org.booking.entity.Flight;
remove unusing import

File: src/main/java/org/booking/Main.java [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/Main.java]: 4 +
remove empty line

File: src/main/java/org/booking/Main.java [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
+import java.time.LocalTime;
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/Main.java]: remove unusing import

File: src/main/java/org/booking/Main.java [hunk 4]
```diff
@@ -1,5 +1,12 @@
...
+import java.util.Calendar;
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/Main.java]: remove unusing import

File: src/main/java/org/booking/Main.java [hunk 5]
```diff
@@ -1,5 +1,12 @@
...
+import java.util.Date;
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/Main.java]: remove unusing import

File: src/main/java/org/booking/Main.java [hunk 6]
```diff
@@ -1,5 +1,12 @@
...
+import java.util.stream.Stream;
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/Main.java]: remove unusing import

File: src/main/java/org/booking/Main.java [hunk 7]
```diff
@@ -1,5 +1,12 @@
...
+
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/Main.java]: remove unusing import

File: src/main/java/org/booking/controllers/FlightController.java [hunk 8]
```diff
@@ -10,16 +10,22 @@
 import org.booking.utils.FileWorker;
 import org.booking.utils.Randomize;
 
+import java.time.LocalDate;
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/controllers/FlightController.java]: remove unusing import

File: src/main/java/org/booking/controllers/FlightController.java [hunk 9]
```diff
@@ -10,16 +10,22 @@
...
+import java.time.LocalTime;
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/controllers/FlightController.java]: remove unusing import

File: src/main/java/org/booking/controllers/FlightController.java [hunk 10]
```diff
@@ -10,16 +10,22 @@
...
 import java.util.ArrayList;
+import java.util.Calendar;
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/controllers/FlightController.java]: remove unusing import

File: src/main/java/org/booking/controllers/FlightController.java [hunk 11]
```diff
@@ -39,9 +46,12 @@ private List<Flight> generateFlights(int count) {
             Airport from = Airport.values()[fromIdx];
             Airport to = Airport.values()[Randomize.num(0, Airport.values().length, fromIdx)];
 
-            flights.add(new Flight(generateTime(), from, to, airline, aircraft, id));
+            flights.add(new Flight(time, from, to, airline, aircraft, id));
+            time += 15 * 60 * 1000;
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/controllers/FlightController.java]: what is this?

File: src/main/java/org/booking/controllers/FlightController.java [hunk 12]
```diff
@@ -39,9 +46,12 @@ private List<Flight> generateFlights(int count) {
...
+            Date date = new Date(time);
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/controllers/FlightController.java]: remove unused variable

File: src/main/java/org/booking/controllers/FlightController.java [hunk 13]
```diff
@@ -39,9 +46,12 @@ private List<Flight> generateFlights(int count) {
...
         }
         return flights;
+
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/controllers/FlightController.java]: remove empty line

**Review by BohdanRadchenko — CHANGES_REQUESTED** [review]
Comments

File: src/main/java/org/booking/controllers/FlightController.java [hunk 14]
```diff
@@ -10,16 +10,22 @@
...
+import java.util.Date;
 import java.util.List;
 
 public class FlightController implements IController {
     private final ServiceFlight service = new ServiceFlight();
 
     private long generateTime() {
-        return System.currentTimeMillis();
+        long currentTime = System.currentTimeMillis();
+        return currentTime / (15 * 60 * 1000) * (15 * 60 * 1000);
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/controllers/FlightController.java]: return currentTime / (15 * 60 * 1000) * (15 * 60 * 1000);

What is the difference between currentTime / (15 * 60 * 1000) * (15 * 60 * 1000) values and currentTime value

**Review by Vladimirov2077 — APPROVED** [review]

File: src/main/java/org/booking/entity/Flight.java [hunk 15]
```diff
@@ -52,4 +52,20 @@ public long getDepartureTimeStamp() {
...
+    public String toString() {
+        String departureTime = String.format("departureTimeStamp=%d", departureTimeStamp);
+        String arrivalTime = String.format("arrivalTimeStamp=%d", arrivalTimeStamp);
+        String depAirport = String.format("departureAirport=%d", departureAirport);
+        String arrAirport = String.format("arrivalAirport=%d", arrivalAirport);
+        String airLine = String.format("airline=%d", airline);
+        String airCraft = String.format("aircraft=%d", aircraft);
+        String fltId = String.format("flightId=%d", flightId);
+        String res = String.format("reserved=%d", reserved);
+        String freeSeat = String.format("freeSeats=%d", freeSeats);
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/entity/Flight.java:65]: String depAirport = String.format("departureAirport=%d", departureAirport);    %d ??
String arrAirport = String.format("arrivalAirport=%d", arrivalAirport);         %d ??
String airLine = String.format("airline=%d", airline);         %d ??
String airCraft = String.format("aircraft=%d", aircraft);         %d ??
String fltId = String.format("flightId=%d", flightId);         %d ??
String fltId = String.format("flightId=%d", flightId);         %d ??
String res = String.format("reserved=%d", reserved);          %d ??

 
 https://docs.oracle.com/javase/7/docs/api/java/util/Formatter.html

File: src/main/java/org/booking/controllers/FlightController.java [hunk 16]
```diff
@@ -7,39 +7,47 @@
 import org.booking.enums.FilePath;
 import org.booking.interfaces.IController;
 import org.booking.services.ServiceFlight;
+import org.booking.utils.Console;
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/controllers/FlightController.java]: remove unused import

File: src/main/java/org/booking/controllers/FlightController.java [hunk 17]
```diff
@@ -7,39 +7,47 @@
...
 import org.booking.utils.FileWorker;
 import org.booking.utils.Randomize;
 
+import java.io.FileNotFoundException;
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/controllers/FlightController.java]: remove unused import

File: src/main/java/org/booking/controllers/FlightController.java [hunk 18]
```diff
@@ -7,39 +7,47 @@
...
 import java.util.ArrayList;
 import java.util.List;
 
 public class FlightController implements IController {
     private final ServiceFlight service = new ServiceFlight();
 
-    private long generateTime() {
-        return System.currentTimeMillis();
+    private List<Long> generateTime(int count) {
```
Comment (BohdanRadchenko) [inline src/main/java/org/booking/controllers/FlightController.java:18]: Now we have DateUtil to work with date. Can you change this method using our utilities?

**Review by BohdanRadchenko — APPROVED** [review]

**Pull request merged**